application/schemas/templates/components.py

```diff
@@ -5,20 +5,25 @@
 from pydantic import Extra
 from pydantic import Field
 from pydantic import constr
+from pydantic import root_validator
 
 from constants.templates import ComponentTypes
+from constants.templates import httpDeploy
+from constants.templates import httpStatus
+from constants.templates import httpDelete
 
 from .validators import TemplateVariable
 
+from typing import Optional
 
 class Component(BaseModel):
```

Comment:
I think you need to split component definition into two(for now) different models. See this [comment](https://github.com/JovianX/Service-Hub/pull/283/files#r1525485019) for more details how to do it. `discriminator` should by `type`.

Reply:
Otherwise it is mess of helm and http components requirements.